client/src/resources/asset.rs

```diff
@@ -253,6 +255,97 @@ mod asset_identifier_tests {
     }
 }
 
+/// When a bad token or string is provided to parsing into an asset
+/// you get an error.
+#[derive(Debug)]
+pub struct ParseAssetError {
```

Comment:
This error type is confusing. We are parsing the asset identifier, not an asset itself, right? If so, this is named incorrectly.